x-sum.py

```diff
@@ -0,0 +1,38 @@
+t= int(input())
+ 
+def x_sum(grid, m, n):
+      
+      dig_sum1= [0 for i in range(n+m-1)]
+      dig_sum2= [0 for i in range(n+m-1)]
+      
+      for i in range(m):
+            for j in range(n):
+                  sum_= i + j
+                  diff= j-i
+                  
+                  dig_sum1[sum_] += grid[i][j]
+                  if diff < 0:
+                        diff= n-1+abs(diff)
```

Comment:
you do not have to do this if you have used dictionary

Reply:
Okay thank you